haskell/src/Stablecoin/Client/Parser.hs

```diff
@@ -278,8 +281,31 @@ clientArgsRawParser = Opt.subparser $
           Nothing
           (#name .! "default-expiry")
           (#help .! "Number of seconds it takes for a permit to expire")
+
+      -- Where to place contract metadata, while keeping it embedded in
+      -- contract being the default
+      dcoContractMetadata <- fromMaybe OpCurrentContract <$> contractMetadataOptParser
       pure $ DeployContractOptions {..}
 
+    contractMetadataOptParser :: Opt.Parser (Maybe ContractMetadataOptions)
+    contractMetadataOptParser =
+        optional (rawOptParser <|> remoteContractOptParser)
```

Comment:
When writing CLI tools, it's good practice to fail when an unexpected/unrecognized option is passed.

The general idea is: it would be very bad if, instead of typing `delete --folder "data"`, I wore to make a typo and typed `delete --flder "data"`, the tool silently ignored the `--flder "data"` option and ended up deleting the current folder's contents instead. So it's best to be really strict and reject all typos/unexpected options.

In our case, there's an unwritten rule that `--contract-metadata-raw` and `contract-metadata-remote` are mutually exclusive: you can use one, or the other, or neither, but never both.

Using `<|>` means if both options are passed, we'll silently ignore the second.
Instead, the parser should fail.

Reply:
When I wrote this, I just assumed that the `opt-parse` library handles this. But now when I look it up in manual, it appears to say this under the 'Alternative' section. I tested it and it indeed seem to reject if both arguments are provided.

> Now --file "foo.txt" will be parsed as FileInput "foo.txt", --stdin will be parsed as StdInput, but a command line containing both options, will be rejected.

Reply:
Oh, didn't know that! That's great then :)